### ISSUE TITLE
Timed-out effects falsely marked as warnings

### ISSUE THREAD

**Issue by Jazzmanpw** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#11]
I have local implementation of `useDebouncedState`, and it does some `setState` in an effect inside a `setTimeout` callback. This is detected by the plugin as a warning, even though it shouldn't be.

```ts
export function useDebouncedState<T>(value: T, delay: number) {
  const [state, setState] = useState(value);
  const [debouncedState, setDebouncedState] = useState(value);

  useEffect(() => {
    const timeout = setTimeout(() => {
      // eslint-disable-next-line react-you-might-not-need-an-effect/you-might-not-need-an-effect
      setDebouncedState(state);
    }, delay);

    return () => {
      clearTimeout(timeout);
    };
  }, [delay, state]);

  return [state, debouncedState, setState] as const;
}
```

I'm thinking about a general rule to skip `setState` calls (and maybe some other checks) in nested callbacks. It doesn't have to be a timeout (or another way to postpone script execution), but I can also initialize some listener and pass a function that might set state or something.

**Comment by NickvanDyke** [comment]
Thanks @Jazzmanpw ! Good catch. This should be fixed in v0.0.39!
